### PR TITLE
Remove Cancellation Switch Journey

### PR DESCRIPTION
**Pull request by andrewHEguardian**
## What does this change?

This PR removes unused code from a Cancellation Product Switch journey which was behind a feature flag for > a year without being turned on, and the associated API definition. 
Also refactors the cancellation container away from AsyncLoader.
Removes a productType definition no longer needed (fixing a ToDo).
Remove incorrect filter string.

## How to test

Deploy this build to code and test cancellation journeys. 

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: shared/productTypes.ts [hunk 1]
```diff
@@ -77,8 +77,7 @@ export type AllProductsProductTypeFilterString =
 	| 'Digipack'
 	| 'SupporterPlus'
 	| 'ContentSubscription'
-	| 'GuardianPatron'
-	| 'RecurringSupport';
```
Comment (charleycampbell) [inline shared/productTypes.ts:81]: quick question why are we removing RecurringSupport?

**Review by charleycampbell — COMMENTED** [review]
A great and much needed refactor!

File: shared/productTypes.ts (hunk 1 above)
Comment (andrewHEguardian) [reply shared/productTypes.ts:81]: Good question. There's a filter function in MDAPI [here](https://github.com/guardian/members-data-api/blob/9efe6e81309ccf90639754d4affd8c8af2d6f5fe/membership-attribute-service/app/services/AccountDetailsFromZuora.scala#L194) which matches on the different strings and `RecurringSupport` is not one of them

File: shared/productTypes.ts [hunk 2]
```diff
@@ -259,7 +258,6 @@ export type ProductTypeKeys =
  */
 export type GroupedProductTypeKeys =
 	| 'membership'
-	| 'contributions'
```
Comment (andrewHEguardian) [inline shared/productTypes.ts:262]: this is where [mdapi defines mmaCategory](https://github.com/guardian/members-data-api/blob/b59dec8c7065a42dd3995907c968d8a382c0efbf/membership-attribute-service/app/models/AccountDetails.scala#L223) and contributions is gone

**Pull request merged**